src/main/java/de/tum/in/www1/artemis/web/rest/competency/PrerequisiteResource.java

```diff
@@ -46,162 +54,301 @@
+     * @param courseId the id of the course for which the prerequisites should be fetched
      * @return the ResponseEntity with status 200 (OK) and with body the found prerequisites
      */
-    @GetMapping("courses/{courseId}/competencies/prerequisites")
-    @EnforceAtLeastStudent
-    public ResponseEntity<List<PrerequisiteResponseDTO>> getPrerequisites(@PathVariable long courseId) {
+    @GetMapping("courses/{courseId}/prerequisites")
+    @EnforceAtLeastStudentInCourse
+    public ResponseEntity<List<Prerequisite>> getPrerequisitesWithProgress(@PathVariable long courseId) {
         log.debug("REST request to get prerequisites for course with id: {}", courseId);
-        Course course = courseRepository.findByIdElseThrow(courseId);
-
-        // Allow any student to see the prerequisites if the course is open to self-enrollment
-        if (!course.isEnrollmentEnabled()) {
-            authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, null);
-        }
-
-        var prerequisites = prerequisiteRepository.findAllByCourseIdOrderById(courseId);
-
-        return ResponseEntity.ok(prerequisites.stream().map(PrerequisiteResponseDTO::of).toList());
+        User user = userRepository.getUserWithGroupsAndAuthorities();
+        final var prerequisites = prerequisiteService.findPrerequisitesWithProgressForUserByCourseId(courseId, user.getId());
+        return ResponseEntity.ok(prerequisites);
     }
 
     /**
-     * GET /courses/:courseId/competencies/prerequisites/:prerequisiteId : Gets the prerequisite competency with the given id for a course.
+     * GET courses/:courseId/prerequisites/:prerequisiteId : gets the prerequisite with the specified id including its related exercises and lecture units
+     * This method also calculates the user progress
      *
-     * @param prerequisiteId the id of the prerequisite to fetch
-     * @param courseId       the id of the course in which the prerequisite should exist
-     * @return the ResponseEntity with status 200 (OK) and with body the found prerequisite or with status 404 (Not Found)
+     * @param prerequisiteId the id of the prerequisite to retrieve
+     * @param courseId       the id of the course to which the prerequisite belongs
+     * @return the ResponseEntity with status 200 (OK) and with body the prerequisite, or with status 404 (Not Found)
      */
-    @GetMapping("courses/{courseId}/competencies/prerequisites/{prerequisiteId}")
+    @GetMapping("courses/{courseId}/prerequisites/{prerequisiteId}")
     @EnforceAtLeastStudentInCourse
-    public ResponseEntity<PrerequisiteResponseDTO> getPrerequisite(@PathVariable long prerequisiteId, @PathVariable long courseId) {
-        log.debug("REST request to get prerequisite with id: {}", prerequisiteId);
+    public ResponseEntity<Prerequisite> getPrerequisite(@PathVariable long prerequisiteId, @PathVariable long courseId) {
```

Comment:
What was the reason for using the `Prerequisite` type directly instead of a DTO?

Reply:
The PrerequisiteResource is currently a copy (with adaptions) of the CompetencyResource. Since that one does not use a DTO for the similar rest call, this one also used no DTO.
In the future we can easily refactor the Resources together and use the CourseCompetencyService to easily create a common DTO for both of them